MarketExecute.php

```diff
@@ -184,7 +195,7 @@ function DefaultErrorHandler($errno, $errstr) {}
             }
             $TypeId = $_POST["TypeId"];
             $Array = array();
-            $Products = ProductsExplorer::GetProductsByType($TypeId);
+            $Products = ProductsExplorer::GetProductsByType($TypeId, $StartId, $ListLength);
```

Comment:
$StartId and $ListLength are required by GetProductsByType and GetProductsByUserId but we are passing them without verifying they are set.

We should either
- Make the 2 parameters optional in ProductExplorer.
- Use CheckPostParam() to check and force user to pass these 2 paramter

I think the first approach makes more sense, pagination should always have a default behavior.